stargz/stargz.go

```diff
@@ -361,17 +367,20 @@ func parentDir(p string) string {
 	return strings.TrimSuffix(dir, "/")
 }
 
-func (r *Reader) getOrCreateDir(d string) *TOCEntry {
+func (r *Reader) getOrCreateDir(d string, dirs map[string]*TOCEntry) *TOCEntry {
 	e, ok := r.m[d]
 	if !ok {
-		e = &TOCEntry{
-			Name: d,
-			Type: "dir",
-			Mode: 0755,
+		e, ok = dirs[d] // expects the directory is in the TOC to preserve the metadata.
```

Comment:
Are these two comment lines part of the same comment? If so, don't use same-line comments mixed with code.